odoo/tools/barcode.py

```diff
@@ -52,3 +53,12 @@ def check_barcode_encoding(barcode, encoding):
            and len(barcode) == barcode_size \
            and re.match(r"^\d+$", barcode) \
            and get_barcode_check_digit(barcode) == int(barcode[-1])
+
+def check_barcode_is_code128(barcode):
+    if barcode:
+        unsupported_chars = re.findall("[^\x00-\x7F\xF1-\xF4]", barcode)
+        if unsupported_chars:
+            error_message = _(
```

Comment:
Does this work? you don't have an environment (so no user language)

Reply:
>Does this work?

Yes, the fix is working properly.
>you don't have an environment (so no user language)

The user entered `Rangée A` this kind of value in the barcode because of which error occurred(`ValueError: Illegal barcode with value 'Rangée A' in code 'Code128'`). So for testing purposes I have copied accented characters from Google and pasted them in barcode value to check the constraint.

Please let me know your suggestions on this.

Reply:
@sauc-odoo I was talking about the translation method. Try to translate the message and to display the error translated.

Reply:
Hello @mart-e 
Changes have been done accordingly, kindly review them.
Thanks.